src/lava/lib/dl/netx/hdf5.py

```diff
@@ -0,0 +1,458 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier:  BSD-3-Clause
+
+"""HDF5 network exchange module."""
+
+from typing import List, Tuple, Union
+from lava.magma.core.decorator import implements
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+import numpy as np
+import h5py
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import InPort, OutPort
+from lava.proc.lif.process import LIF
+from lava.proc.sdn.process import Sigma, Delta, SigmaDelta
+from lava.lib.dl.netx.utils import NetDict
+from lava.lib.dl.netx.utils import optimize_weight_bits
+from lava.lib.dl.netx.blocks.process import Input, Dense, Conv
+from lava.lib.dl.netx.blocks.models import AbstractPyBlockModel
+
+
+class Network(AbstractProcess):
+    """Generates a Lava process for the network described in hdf5 config.
+
+    Parameters
+    ----------
+    net_config : str
+        name of the hdf5 config file.
+    num_layers : int, optional
+        number of blocks to generate. An integer valuew will only generate the
+        first ``num_layers`` blocks in the description. The actual number of
+        generated layers may be less than ``num_layers``. If it is None, all
```

Comment:
Should filename be listed in the params?

Reply:
net_config is the filename string. Perhaps file was misleading. Changed it to filename in the updated docstring.